src/runtime/agas/interface.cpp

```diff
@@ -22,10 +22,17 @@
 
 namespace hpx { namespace agas
 {
+// request the reference to the AGAS client only once
+naming::resolver_client& get_agas_client()
+{
+    static naming::resolver_client& client = naming::get_agas_client();
```

Comment:
@hkaiser This doesn't play well with the tests that initialize the runtime multiple times (the ones that are failing on CircleCI). Does this change make a big difference?

Reply:
@msimberg Good point! I forgot about the multiple-init scenarios. I'll roll this change back. This could also explain the hangs. Thanks!